src/extension/infer.rs

```diff
@@ -516,10 +516,12 @@ impl UnificationContext {
                                 solved = true;
                             }
                         };
-                    } else if let Some(superset) = self.get_solution(&meta) {
-                        let subset = ExtensionSet::singleton(r).missing_from(superset);
-                        self.add_solution(self.resolve(*other_meta), subset);
-                        solved = true;
+                    } else if let Some(_superset) = self.get_solution(&meta) {
+                        // Here, we're stuck because the Plus constraint only
+                        // signifies a preorder. I.e. if m0 = m1 + 'R', it's
+                        // still possible that the solution is m0 = m1 because
+                        // it's possible that we're adding 'R' to a set which
+                        // already contained it.
```

Comment:
Either just add `So, do nothing.` to make it clear this `if` here is just to scope the comment; Or, comment out the `else if` as well. Of course that'll then become noncompiling and soon break, so if you go that way, might be better to translate it from rust to english ;)